src/Storages/StorageSet.h

```diff
@@ -64,6 +68,7 @@ class StorageSetOrJoinBase : public IStorage
   *  and also written to a file-backup, for recovery after a restart.
   * Reading from the table is not possible directly - it is possible to specify only the right part of the IN statement.
   */
+template <bool is_prob>
```

Comment:
By the way if we will go with exacting interface `ISet` and two different implementations `class Set : public ISet` and `class ProbSet : public ISet` this parameter will be a runtime parameter passed to `StorageSet` constructor